pkgs/tools/system/plan9port/builder.sh

```diff
@@ -3,10 +3,26 @@ source $stdenv/setup
 export PLAN9=$out/plan9
 export PLAN9_TARGET=$PLAN9
 
+plan9portLinkFlags()
+{
+    local -a linkFlags=()
+    eval set -- "$NIX_LDFLAGS"
+    while (( $# > 0 )); do
+        if [[ $1 = -rpath ]]; then
+            linkFlags+=( "-Wl,-rpath" "-Wl,$2" )
+            shift 2
+        else
+            linkFlags+=( "$1" )
+            shift
+        fi
+    done
+    echo "${linkFlags[*]}"
+}
```

Comment:
I’m curious what what plan9port specifically means it needs rpaths to be `-Wl`… Presumably other packages don’t need this?

Reply:
plan9port wants to use the C compiler as a linker.  Apparently passing `-rpath` to the C compiler works fine for clang, but `-Wl,-rpath` is required for gcc, so this fails on NixOS with a million "bad ELF type" messages - presumably because it's trying to link the rpath directory as an object file?

If it attempted to use `ld` instead of `gcc` (which seems more sane), this would be fine.